yarn-project/aztec-nr/aztec/src/note/lifecycle.nr

```diff
@@ -30,6 +35,24 @@ pub fn create_note<Note, N>(
     assert(notify_created_note(storage_slot, preimage, inner_note_hash) == 0);
 
     context.push_new_note_hash(inner_note_hash);
+
+    let serialize = note_interface.serialize;
+
+    if broadcast {
+        let get_owner = note_interface.get_owner;
+        let maybe_owner = get_owner(*note);
+        assert(maybe_owner.is_some(), "Note owner is must be set when broadcast flow is triggered.");
+
+        let owner = maybe_owner.unwrap_unchecked();
+        let owner_key = get_public_key(owner);
+        emit_encrypted_log(
```

Comment:
Something we might want to consider for an upcoming change on notes generally could be to put more of this logic into the note, and have the lifecycle just call `note.emit_encrypted_log()` or something like that? It puts more stuff in the hands of the dev, but makes it a bit more clear what is put into the log etc 🤷. Probably just that I'm not too fond of the way we end up calling back and forth as I think it makes it harder to follow what happens.